evalml/pipelines/utils.py

```diff
@@ -1486,37 +1489,61 @@ def stack_X(X, series_id_name, time_index, starting_index=None):
         time_index (str): The name of the time index column.
         starting_index (int): The starting index to use for the stacked DataFrame. If None, the starting index
             will match that of the input data. Defaults to None.
+        series_id_values (set): The unique values of a series ID, used to generate the index. If None, values will
+            be generated from X column values. Required if X only has time index values and no exogenous values.
+            Defaults to None.
 
     Returns:
         pd.DataFrame: The restacked features.
     """
     original_columns = set()
-    series_ids = set()
-    for col in X.columns:
-        if col == time_index:
-            continue
-        separated_name = col.split("_")
-        original_columns.add("_".join(separated_name[:-1]))
-        series_ids.add(separated_name[-1])
-
-    restacked_X = []
-
-    for i, original_col in enumerate(original_columns):
-        # Only include the series id once (for the first column)
-        include_series_id = i == 0
-        subset_X = [col for col in X.columns if original_col in col]
-        restacked_X.append(
-            stack_data(
-                X[subset_X],
-                include_series_id=include_series_id,
-                series_id_name=series_id_name,
-                starting_index=starting_index,
-            ),
+    series_ids = series_id_values or set()
+    if series_id_values is None:
+        for col in X.columns:
+            if col == time_index:
+                continue
+            separated_name = col.split("_")
+            original_columns.add("_".join(separated_name[:-1]))
+            series_ids.add(separated_name[-1])
+
+    if len(series_ids) == 0:
+        raise ValueError(
+            "Series ID values needs to be passed in X column values or as a set with the `series_id_values` parameter.",
         )
-    restacked_X = pd.concat(restacked_X, axis=1)
 
     time_index_col = X[time_index].repeat(len(series_ids)).reset_index(drop=True)
-    time_index_col.index = restacked_X.index
-    restacked_X[time_index] = time_index_col
+
+    if len(original_columns) == 0:
+        start_index = starting_index or X.index[0]
+        stacked_index = pd.RangeIndex(
+            start=start_index,
+            stop=start_index + len(time_index_col),
+        )
+        time_index_col.index = stacked_index
```

Comment:
I don't think we need this line

Reply:
We need it in the case where the `time_index_col` has a different starting index value other than 0. Without it, the test cases where `starting_index` is not None will fail.